api/src/modules/contextual-layers/contextual-layer.entity.ts

```diff
@@ -0,0 +1,60 @@
+import {
+  BaseEntity,
+  Column,
+  Entity,
+  ManyToOne,
+  PrimaryGeneratedColumn,
+} from 'typeorm';
+import { BaseServiceResource } from 'types/resource.interface';
+import { ApiProperty, ApiPropertyOptional } from '@nestjs/swagger';
+import { H3Data } from 'modules/h3-data/h3-data.entity';
+
+export enum CONTEXTUAL_LAYER_CATEGORY {
+  ENVIRONMENTAL_DATASETS = 'Environmental datasets',
+  BUSINESS_DATASETS = 'Business datasets',
+  FOOD_AND_AGRICULTURE = 'Food and agriculture',
+  DEFAULT = 'Default',
+}
+
+export const businessUnitResource: BaseServiceResource = {
+  className: 'ContextualLayer',
+  name: {
+    singular: 'contextualLayer',
+    plural: 'contextualLayers',
+  },
+  entitiesAllowedAsIncludes: [],
+  columnsAllowedAsFilter: ['name', 'metadata', 'category'],
+};
+
+@Entity()
+export class ContextualLayer extends BaseEntity {
+  @PrimaryGeneratedColumn('uuid')
+  @ApiProperty()
```

Comment:
Small question - since there is no controller and no endpoint for user to interact directly with that entity (so it won't be in swagger ui, if I understood correctly), do we need the openAPI tags like @ApiProperty() here?

Reply:
Good catch. We don't need a controller now since this task is to allow other people to actually use this new entity, but once we have clear requirements we will be using a controller for this module soon enough